File: www/cgi-bin/handler.php
#!/usr/bin/php-cgi
<?php
header("Content-Type: text/html");
echo "<html><body><h1>PHP POST Handler</h1>";

if ($_SERVER['REQUEST_METHOD'] === 'POST' && isset($_POST['name'])) {
    $name = htmlspecialchars($_POST['name']);
    echo "<p>Hello, <strong>$name</strong>!</p>";
} else {
    echo "<p>No name was submitted.</p>";
}

echo "<a href='/'>Go back</a>";
echo "</body></html>";
?>

File: www/cgi-bin/handler.py
#!/usr/bin/env python3
import sys
import cgi

print("Content-Type: text/html\r\n\r\n")
print("<html><body><h1>Python POST Handler</h1>")

form = cgi.FieldStorage()
if "name" in form:
    name = form.getvalue("name")
    print(f"<p>Hello, <strong>{name}</strong>!</p>")
else:
    print("<p>No name was submitted.</p>")
    
print("<a href='/'>Go back</a>")
print("</body></html>")